### ISSUE TITLE
Must run make dep twice on Go 1.16

### ISSUE THREAD

**Issue by okuuva** [marekaf/gcr-lifecycle-policy#18]
I was integrating gcr-lifecycle-policy to a Docker image while I ran into a problem with the latest official golang image. Turns out that with 1.16 image one must run `make dep` twice, otherwise build fails with the following error message:
```
#6 15.14 go: downloading golang.org/x/lint v0.0.0-20201208152925-83fdc39ff7b5
#6 15.47 go: downloading golang.org/x/tools v0.0.0-20200130002326-2f3ba24bd6e7
#6 16.69 go: downloading golang.org/x/tools v0.1.0
#6 19.28 go get: upgraded golang.org/x/lint v0.0.0-20190313153728-d0100b6bd8b3 => v0.0.0-20201208152925-83fdc39ff7b5
#6 19.28 go get: upgraded golang.org/x/net v0.0.0-20200822124328-c89045814202 => v0.0.0-20201021035429-f5854403a974
#6 19.28 go get: upgraded golang.org/x/tools v0.0.0-20190312170243-e65039ee4138 => v0.1.0
#6 19.78 ../pkg/mod/golang.org/x/oauth2@v0.0.0-20190604053449-0f29369cfe45/internal/token.go:23:2: missing go.sum entry for module providing package golang.org/x/net/context/ctxhttp; to add:
#6 19.78        go mod download golang.org/x/net
#6 19.78 ../pkg/mod/k8s.io/apimachinery@v0.0.0-20190913080033-27d36303b655/pkg/util/net/http.go:34:2: missing go.sum entry for module providing package golang.org/x/net/http2; to add:
#6 19.78        go mod download golang.org/x/net
#6 19.78 ../pkg/mod/github.com/sirupsen/logrus@v1.6.0/terminal_check_unix.go:6:8: missing go.sum entry for module providing package golang.org/x/sys/unix; to add:
#6 19.78        go mod download golang.org/x/sys
#6 19.78 make: *** [Makefile:16: build] Error 1
------
executor failed running [/bin/sh -c wget -qO - "https://github.com/marekaf/gcr-lifecycle-policy/archive/${GCR_VERSION}.tar.gz"  | tar --strip-components=1 -xzf -  && make dep && make build]: exit code: 2
```

As said, running `make dep` twice fixes the problem.
Here's the Dockerfile and build commands for reference:

```dockerfile
ARG GOLANG_VERSION=1.15
FROM golang:$GOLANG_VERSION as build-gcr

WORKDIR /go/src

ENV CGO_ENABLED=0
ARG GCR_VERSION=0.2

# For some reason `make dep` doesn't pull all the dependencies in one go, must be run twice
RUN wget -qO - "https://github.com/marekaf/gcr-lifecycle-policy/archive/${GCR_VERSION}.tar.gz" \
 | tar --strip-components=1 -xzf - \
 && make dep && make build
```

`docker build --pull --build-arg GOLANG_VERSION=1.15 --tag gcr-build:1.15 .` Works
`docker build --pull --build-arg GOLANG_VERSION=1.16 --tag gcr-build:1.16 .` Fails unless another `&& make dep` is added.

**Comment by marekaf** [comment]
Hi, thanks for submitting the issue!

I think I managed to fix it https://github.com/marekaf/gcr-lifecycle-policy/commit/de9525a4be3751fca8ca78701c8237c6f5f38433 

I released a new version `0.3`. Try to bump the `GCR_VERSION=0.3` and it should work with just single make dep (or even without it!)

**Comment by okuuva** [comment]
Yup, working now with just a single `make dep`. Thanks for a quick response and fix!
